internal/operator/backrest/repo.go

```diff
@@ -218,3 +300,13 @@ func createService(clientset kubernetes.Interface, fields *RepoServiceTemplateFi
 
 	return err
 }
+
+// S3RepoTypeCLIOptionExists detects if a S3 restore was requested via the '--repo-type'
+// command line option
+func S3RepoTypeCLIOptionExists(opts string) (exists bool, err error) {
+	exists, err = regexp.MatchString(S3RepoTypeRegexString, opts)
```

Comment:
The `err` here comes from parsing `S3RepoTypeRegexString`. I recommend unexporting the constant and precompiling it. Not a blocker.

```go
var regexpS3RepoType = regexp.MustCompile(`--repo-type=["']?s3["']?`)

func S3RepoTypeCLIOptionExists(opts string) bool {
  return regexpS3RepoType.MatchString(opts)
}
```

🤔 &nbsp;I only see this function called once, so maybe it should be unexported as well.

Reply:
Its actually called two places.  Within this same file (`repo.go`) and then again here:
https://github.com/CrunchyData/postgres-operator/pull/1679/files#diff-c86fbc1d6fff5e4151b2cbedca8a3f65R2148

Reply:
Thanks, updated per recommendation.